barman/config.py

```diff
@@ -1676,6 +1687,317 @@ def _validate_with_keys(config_items, allowed_keys, section):
                 )
 
 
+class BaseChange:
+    """
+    Base class for change objects.
+
+    Provides methods for equality comparison, hashing, and conversion
+    to tuple and dictionary.
+    """
+
+    _fields = []
+
+    def __eq__(self, other):
+        """
+        Equality support.
+        """
+        if isinstance(other, self.__class__):
+            return self.as_tuple() == other.as_tuple()
+        return False
+
+    def __hash__(self):
+        """
+        Hash/set support.
+        """
+        return hash(self.as_tuple())
+
+    def as_tuple(self) -> tuple:
+        """
+        Convert to a tuple, ordered as self._fields.
+        """
+        return tuple(vars(self)[k] for k in self._fields)
+
+    def as_dict(self):
+        """
+        Convert to a dictionary, using self._fields as keys.
+        """
+        return {k: vars(self)[k] for k in self._fields}
+
+
+class ConfigChange(BaseChange):
+    """
+    Represents a configuration change received.
+
+    Attributes:
+        key (str): The key of the configuration change.
+        value (str): The value of the configuration change.
+        config_file (str, optional): The configuration file associated with the change.
```

Comment:
```suggestion
    :ivar key str: The key of the configuration change.
    :ivar value str: The value of the configuration change.
    :ivar config_file Optional[str]: The configuration file associated with the change, or ``None``.
```

Use reST, which is used through the code, instead of Google style docstrings.